apps/next/components/Waitlist.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+import { joinWaitlist } from "@/app/actions";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { ServerActionState, serverActionInitialState } from "@/lib/action";
+import { cn } from "@/lib/utils";
+import { useEffect, useState } from "react";
+import { useFormState, useFormStatus } from "react-dom";
+import confetti from "canvas-confetti";
+
+function Waitlist() {
+  const [state, formAction] = useFormState(
+    joinWaitlist,
+    serverActionInitialState
+  );
+  const [email, setEmail] = useState("");
+
+  useEffect(() => {
+    if (state.message.includes("You're on the waitlist!")) {
```

Comment:
If you accept my suggestion to change the on-join-waitlist message, don't forget to update it here too.

Reply:
👍